firestore/options.go

```diff
@@ -61,12 +61,8 @@ func LastUpdateTime(t time.Time) Precondition { return lastUpdateTime(t) }
 type lastUpdateTime time.Time
 
 func (u lastUpdateTime) preconditionProto() (*pb.Precondition, error) {
-	ts, err := ptypes.TimestampProto(time.Time(u))
-	if err != nil {
-		return nil, err
-	}
```

Comment:
I'm *guessing* this does some validation of the time, but I could be wrong. Is it worth using the `Timestamp.IsValid` api to retain that?

Could be wrong though.

Reply:
added back with checks